### PR TITLE
Initial GPU e2e test

### PR DESCRIPTION
**Pull request by njuettner**
Tests if a vector job succeeds on a GPU nodes

### CODE REVIEW

File: test/e2e/cluster_config.sh [hunk 1]
```diff
@@ -93,6 +93,16 @@ clusters:
     profile: ${WORKER_PROFILE}-default
     min_size: 1
     max_size: 21
+  - discount_strategy: spot_max_price
+    instance_types: ["p3.2xlarge", "g2.2xlarge", "g3s.xlarge", "g3.4xlarge"]
+    name: worker-gpu
+    profile: ${WORKER_PROFILE}-default
+    min_size: 1
```
Comment (aermakov-zalando) [inline test/e2e/cluster_config.sh]: Let's go with `min_size: 0`? Then we can also test the autoscaling.

File: test/e2e/gpu.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+		_, err = cs.CoreV1().Pods(ns).Get(pod.Name, metav1.GetOptions{})
+		if err != nil {
+			Expect(fmt.Errorf("Could not get POD %s", pod.Name)).NotTo(HaveOccurred())
+			return
+		}
+
+		f.PodClient().WaitForSuccess(pod.Name, 10*time.Minute)
+		logs, err := getPodLogs(cs, ns, pod.Name, "cuda-vector-add", false)
+		framework.ExpectNoError(err, "Should be5ble to get logs for pod %v", pod.Name)
```
Comment (aermakov-zalando) [inline test/e2e/gpu.go]: "be able"?

File: test/e2e/gpu.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+		By("Creating a vector pod which runs on a GPU node")
+		pod := createVectorPod(nameprefix, ns, labels)
+		_, err := cs.CoreV1().Pods(ns).Create(pod)
+		Expect(err).NotTo(HaveOccurred())
+		framework.ExpectNoError(f.WaitForPodRunning(pod.Name))
+		for {
+			p, err := cs.CoreV1().Pods(ns).Get(pod.Name, metav1.GetOptions{})
+			if err != nil {
+				Expect(fmt.Errorf("Could not get POD %s", pod.Name)).NotTo(HaveOccurred())
```
Comment (aryszka) [inline test/e2e/gpu.go]: this might be better:

```
framework.ExpectNoError(err, "...
```

File: test/e2e/gpu.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+				return
+			}
+			if p.Status.ContainerStatuses[0].State.Terminated == nil {
+				time.Sleep(10 * time.Second)
+				continue
+			}
+			n := p.Status.ContainerStatuses[0].State.Terminated.ExitCode
+			if n < 1 {
```
Comment (aryszka) [inline test/e2e/gpu.go]: should the test fail if the exit code is not zero?

Comment (njuettner) [reply test/e2e/gpu.go]: I added this case below. Shouldn't be the case but might not harm adding it. Thanks!

**Comment by njuettner** [comment]
👍

**Comment by mikkeloscar** [comment]
:+1:

**Pull request merged**